frontend/src/components/ShoppingLists.tsx

```diff
@@ -0,0 +1,65 @@
+import {useState} from "react";
+import {useNavigate} from "react-router-dom";
+
+type ShoppingListsProps = {
+    lists: {
+        id: string,
+        name: string
+    }[],
+    //onButtonClick: (item: { id: string, name: string }) => void
+}
+
+export default function ShoppingLists ({lists}: ShoppingListsProps)  {
+
+    const [searchTerm, setSearchTerm] = useState('');
+
+    const filteredList = lists.filter((shoppingList) =>
+        shoppingList.name.toLowerCase().includes(searchTerm.toLowerCase()));
+
+    const handleGoShoppingButtonClick = (id: string) => {
+        navigate("/shopping/" + id)
+    };
+
+    const handleNewShoppingListButtonClick = () => {
+        navigate("/shoppinglist")
+    };
+
+    const navigate = useNavigate();
```

Comment:
The navigate declaration should appear before its usage, so you could move it up to line 18.